File: src/main/java/com/sparkTutorial/rdd/nasaApacheWebLogs/SameHostsSolution.java
package com.sparkTutorial.rdd.nasaApacheWebLogs;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

public class SameHostsSolution {

    public static void main(String[] args) throws Exception {

        SparkConf conf = new SparkConf().setAppName("sameHosts").setMaster("local[1]");

        JavaSparkContext sc = new JavaSparkContext(conf);

        JavaRDD<String> julyFirstLogs = sc.textFile("in/nasa_19950701.tsv");
        JavaRDD<String> augustFirstLogs = sc.textFile("in/nasa_19950801.tsv");

        JavaRDD<String> julyFirstHosts = julyFirstLogs.map(line -> line.split("\t")[0]);

        JavaRDD<String> augustFirstHosts = augustFirstLogs.map(line -> line.split("\t")[0]);

        JavaRDD<String> intersection = julyFirstHosts.intersection(augustFirstHosts);

        JavaRDD<String> cleanedHostIntersection = intersection.filter(host -> !host.equals("host"));

        cleanedHostIntersection.saveAsTextFile("out/nasa_logs_same_hosts.csv");
    }
}

File: src/main/scala/com/sparkTutorial/rdd/nasaApacheWebLogs/SameHostsSolution.scala
package com.sparkTutorial.rdd.nasaApacheWebLogs

import org.apache.spark.SparkConf
import org.apache.spark.api.java.JavaSparkContext

object SameHostsSolution {

    def main(args: Array[String]) {

        val conf = new SparkConf().setAppName("sameHosts").setMaster("local[1]");

        val sc = new JavaSparkContext(conf)

        val julyFirstLogs = sc.textFile("in/nasa_19950701.tsv")
        val augustFirstLogs = sc.textFile("in/nasa_19950801.tsv")

        val julyFirstHosts = julyFirstLogs.map(line => line.split("\t")(0))
        val augustFirstHosts = augustFirstLogs.map(line => line.split("\t")(0))

        val intersection = julyFirstHosts.intersection(augustFirstHosts)

        val cleanedHostIntersection = intersection.filter(host => host != "host")
        cleanedHostIntersection.saveAsTextFile("out/nasa_logs_same_hosts.csv")
    }
}
